setup.py

```diff
@@ -8,6 +8,27 @@
 
 
 python_requires = "~=3.6"
+setup_requires = ["setuptools_scm"]
+install_requires = ["typing_extensions>=3.6"]
+dev_requires = [
+    "black",
+    "codecov",
+    "coverage[toml]",
+    "flake8",
+    "flake8-bugbear",
+    "flake8-builtins",
+    "flake8-comprehensions",
+    "flake8-i18n",
+    "invoke",
+    "isort",
+    "mypy",
+    "pip-tools",
+    "py-githooks",
+    "pytest",
+    "semver",
+    "twine",
+    "wheel",
```

Comment:
What's the reasoning behind removing the version requirements?

Reply:
The main purpose is to remove the duplication and fragmentation. The version pinning for dev dependencies (running tests) are not critical compared to the distribution requirements (e.g. typing_extensions)

Reply:
We could pin them in here as well, just didn't think it was necessary